### ISSUE TITLE
[Notebooks] Diagnostics are not removed when language is changed

### ISSUE THREAD

**Issue by claudiaregio** [microsoft/pylance-release#3555]
## Steps to reproduce: 
1. Open a Polyglot Notebook 
2. Sometimes kernel is set to a Python kernel by default
3. Switch kernel to .NET Interactive
4. Pylance diagnostics do not go away once the correct kernel is assigned to notebook

Below shows a notebook where .NET Interactive has been set as the kernel for Polyglot notebooks, yet Pylance is still claiming there are errors on a cell that is actually C# code, not Python. 

![image](https://user-images.githubusercontent.com/19276747/198911531-223d6a3f-4da7-436d-8ba7-c13864c5033a.png)

**Comment by rchiodo** [comment]
Looking at the LSP messages, it seems we don't clear diagnostics when the notebook close event comes?

I see this when I switch languages:

```json
[Trace - 9:53:10 AM] Sending notification 'notebookDocument/didClose'.
Params: {
    "notebookDocument": {
        "uri": "untitled:Untitled-1.ipynb?jupyter-notebook"
    },
    "cellTextDocuments": [
        {
            "uri": "vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D"
        }
    ]
}
```

**Comment by debonte** [comment]
Related to https://github.com/microsoft/pyrx/pull/2873. Closing a notebook cell means that it is deleted and we should remove all info about it. Which is different from closing a document which is temporary -- the user could immediately reopen it.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221121-9-November-2022-PreRelease)
